### DISCUSSION TITLE
How can I delete indexed repositories for GitHub Copilot in GitHub.com?

### DISCUSSION THREAD

**Discussion by zavocc** [community/community#142387] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have individual subscription and I have been testing the indexing capability that Copilot in GitHub.com offers, although I ran out of repository indexes which is 5 and some test repos I have is indexed to test the feature, I couldn't seem to find a setting or an option to stop the repository from being Indexed so that I can index other repos... I believe the only way to remove the index is to delete the repository altogether which is highly impractical

Would be helpful if there is such an option to do so, if there are workarounds, please let us know, thanks in advance!

**Comment by qkfang** [comment]
Yup, you are correct.

Warning: Indexing is permanent for each repository unless the repository itself is deleted. Use caution when selecting a repository to index. Repository indexing cannot be canceled once it starts.

https://docs.github.com/en/copilot/managing-copilot/managing-github-copilot-in-your-organization/customizing-copilot-for-your-organization/indexing-repositories-for-copilot-chat

**Reply by raptor-07** [reply]
That is not a serious solution at all. What if I indexed a FOSS repository to understand the codebase? Will the index at least be useful for other contributors?

**Comment by kimasplund** [comment]
Came here for this solution also. This seems a bit ... Stupid. 

So i need to delete the repo in order to be able to index another repo? 
Every time i think i should use copilot it finds new ways to disappoint me...

**Comment by johnkennyuk** [comment]
Same here, I haven't tired deleting a repo of mine yet so I don't yet know if this releases repo indexing slots? I sure hope that's how it works especially since the docs suggest deleting repos is the only way to delete repo indexes, yet it doesn't state whether that then releases used indexing slots or not??

If there persistent and the indexing limits are lifetime limits I won't be happy lol!!!

**Comment by leobalter** [comment]
> although I ran out of repository indexes which is 5 (...)

Thanks for the feedback, we're working on a fix for this and will have an update pretty soon on this matter.

**Reply by gmagana** [reply]
> > although I ran out of repository indexes which is 5 (...)
> 
> Thanks for the feedback, we're working on a fix for this and will have an update pretty soon on this matter.

Can you define "pretty soon"?

Seriously, this UX sucks, much more since we are paying users.

**Reply by leobalter** [reply]
Hi @gmagana, I'm sorry for not posting an update earlier. 

Since the [Copilot Chat GA](https://github.blog/changelog/2024-12-18-copilot-chat-on-github-is-now-generally-available-for-all-users/) release, there is no limit to how many repositories you can index. This applies to all subscription levels, including GitHub Copilot Free.

**Comment by Elli86New** [comment]
??? same

**Comment by kimasplund** [comment]
Any update or progress on this little "inconvenience"? 

honestly i dont even bother opening Copilot anymore :-/

**Comment by zavocc** [comment]
https://github.blog/changelog/2024-12-18-copilot-chat-on-github-is-now-generally-available-for-all-users/

Makes me wondering if this issue is still relevant now as Indexing i
believe is unlimited and enabled by default to all repos?

**Reply by leobalter** [reply, marked as answer]
yes, since the [Copilot Chat GA](https://github.blog/changelog/2024-12-18-copilot-chat-on-github-is-now-generally-available-for-all-users/) - as you pointed out in the link - [indexing is no longer limited](https://docs.github.com/en/copilot/using-github-copilot/indexing-repositories-for-copilot-chat). When you open the Copilot Chat at GitHub, the indexing is automatically triggered. The automatic indexing should be available soon in IDEs, including VSCode.

The unlimited indexing applies to all subscription levels, including GitHub Copilot Free. I hope you enjoy this new experience with better context-accurate responses from Copilot.

Ping me anytime for comments, questions, and feedback!

**Reply by TotoShampoin** [reply]
@leobalter 
I have one question

If I accidentally clicked to index a repo that I did not want to index, I cannot cancel that, right?
Like, I don't care whether it is limited or not, I'm just generally concerned about the fact that I did something by accident that I cannot revert

Question is Why can't it be reverted? Does Copilot trains itself with that?

**Reply by eltafhussain** [reply]
I am facing this same issue, I am can't disable the indexing for a repo after I accidentally clicked the indexing option, though it is very helpful feature, but for cases where we no longer require indexing or want to remove the indexed repo, there is no option that I can find or aware to disabled or remove it.

**Reply by eltafhussain** [reply]
@leobalter can you help us out here, if there is a way to stop already triggered indexing option?

**Reply by leobalter** [reply]
Hi @eltafhussain, no worries! To clarify the key points here:

Copilot does not use the semantic indexing for model training. All the indexing does is making Copilot ready to perform semantic searches on your repo's code if that information can be useful to answer your questions.

The performed search is equivalent - if not the same - to anyone using the search bar on GitHub.com.

The indexed data remains within your GitHub environment and is only accessible to users who already have access to that repo.

Thanks for your feedback!

-Leo

**Reply by TotoShampoin** [reply]
@leobalter 

Even if there is nothing to worry about, I still think it's stupid to not provide the users an option to revert it. You know, give the users a sense of freedom, not forcing them to deal with the consequences of an accident.

But that's not the Microsoft way, is it

**Reply by kingmalitha** [reply]
@leobalter hi, currently I have a problem, I am working on medium size nestjs application. However it seems like because of the index, the autocompletion (Intellisense and other extension) are becomes slower. The gitHub copilot extension itself takes around 5000 ms + time to activate.

As a result, I can not work on that repo with GitHub copilot, so I have to disable it, in order to get the intellisense working realtime.

What can I for this scenario ??

I tried disabling the extension, then I got faster intellisense. Note this issue only happen only one workspace. Other workspace works fine.
